src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -155,20 +173,22 @@ export class QuickOpenDialog extends LitElement {
   ): CommandItem[] {
     return items
       .filter(({ text }) =>
-        fuzzySequentialMatch(filter.toLowerCase(), text.toLowerCase())
+        fuzzySequentialMatch(filter.toLowerCase(), [text.toLowerCase()])
       )
       .sort((itemA, itemB) => compare(itemA.text, itemB.text));
   }
 
-  private _filterEntityItems(
-    entity_ids: HassEntity["entity_id"][],
-    filter: string
-  ): HassEntity["entity_id"][] {
-    return entity_ids
-      .filter((entity_id) =>
-        fuzzySequentialMatch(filter.toLowerCase(), entity_id)
+  private _filterEntityItems(filter: string): HassEntity[] {
+    return this._entities
+      .filter((entity) =>
+        fuzzySequentialMatch(filter.toLowerCase(), [
+          entity.entity_id,
+          entity.attributes?.friendly_name?.toLowerCase() || "",
```

Comment:
If it doesn't have a friendly name we will still try to match it now, we should not include it in the array instead.

Reply:
It would send an empty string in that case, which make the sequence matcher immediately exit:

https://github.com/home-assistant/frontend/blob/dev/src/common/string/sequence_matching.ts#L11

But I went ahead and pushed a change to keep it from ever sending.